pkg/resources/resources.go

```diff
@@ -631,3 +645,95 @@ func ListAvailableAPIResources(
 
 	return items, nil
 }
+
+// DeleteResources iterates through a list of ResourceSpec and deletes matching Kubernetes resources.
+// It collects all errors encountered during deletion and returns them as a multierror.
+//
+// Parameters:
+//   - ctx: The context for the operation
+//   - c: The Kubernetes client used to list and delete resources
+//   - resources: A pointer to a slice of ResourceSpec defining which resources to delete
+//
+// Returns:
+//   - error: A multierror containing all errors encountered, or nil if all deletions succeeded
+func DeleteResources(ctx context.Context, c client.Client, resources *[]ResourceSpec) error {
+	var errors *multierror.Error
+
+	for _, res := range *resources {
```

Comment:
Why the pointer? This could be changed to 

```suggestion
func DeleteResources(ctx context.Context, c client.Client, resources []ResourceSpec) error {
	var errors *multierror.Error

	for _, res := range resources {
```